src/server.cpp

```diff
@@ -3797,6 +3798,9 @@ PlayerSAO* Server::emergePlayer(const char *name, session_t peer_id, u16 proto_v
 	playersao->finalize(player, getPlayerEffectivePrivs(player->getName()));
 	player->protocol_version = proto_version;
 
+	// Set player lang_code
+	player->m_lang_code = getClient(peer_id, CS_Invalid)->m_lang_code;
```

Comment:
please add a getter on the m_lang_code instead of calling the variable publically